Make Atlas connectivity tests more robust

divjotarora opened the pull request:
I've seen these tests fail sometimes with because the inserted document isn't found, so I added write concern majority and read concern local (I tried read concern majority but I believe some of these are running against older servers that weren't started with `--enableMajorityReadConcern`).

Also, the current tests use `log.Fatalf` which calls `os.Exit` in a failure. This means that `defer` functions do not run, so the created client isn't closed in failure cases. I changed these to `panic` instead and added the URI index to the error messages so we can easily find the cluster URI if we have failures that aren't transient.

jyemin commented:
I'm surprised by how much this test is doing.  DRIVERS-382 calls for just an `ismaster` and a `findOne` (but I wouldn't assume there's a document in the collection, so expect nil.  Java driver actually just does an `ismaster` and an `estimatedDocumentCount`) and only asserts that neither of them error.

divjotarora commented:
@jyemin I didn't know that. Not sure why the `InsertOne` was there, but I removed it and changed the error checking to check for non-nil and not `ErrNoDocuments`.
